tests/foreman/ui/test_audit.py

```diff
@@ -0,0 +1,258 @@
+                    'entity': entities.LibvirtComputeResource(),
+                    'entity_type': 'compute_resource',
+                    'value_template': '{entity.name} (Libvirt)'
+                },
+                {
+                    'entity': entities.ConfigGroup(),
+                    'entity_type': 'config_group'
+                },
+                {'entity': entities.Domain(), 'entity_type': 'domain'},
+                {'entity': entities.Host(), 'entity_type': 'host'},
+                {'entity': entities.HostGroup(), 'entity_type': 'hostgroup'},
+                {'entity': entities.Image(
+                    compute_resource=entities.LibvirtComputeResource().create()
+                ), 'entity_type': 'image'},
+                {'entity': entities.Location(), 'entity_type': 'location'},
+                {
+                    'entity': entities.Media(),
+                    'entity_type': 'medium',
+                    'custom_operation': 'added',
+                },
+                {
+                    'entity': entities.Organization(),
+                    'entity_type': 'organization'
+                },
+                {
+                    'entity': entities.OperatingSystem(),
+                    'entity_type': 'os',
+                    'value_template': '{entity.name} {entity.major}'
+                },
+                {
+                    'entity': entities.PartitionTable(),
+                    'entity_type': 'ptable',
+                },
+                {
+                    'entity': entities.PuppetClass(),
+                    'entity_type': 'puppetclass'
+                },
+                {'entity': entities.Role(), 'entity_type': 'role'},
+                {
+                    'entity': entities.Subnet(),
+                    'entity_type': 'subnet',
+                    'value_template': '{entity.name} '
+                                      '({entity.network}/{entity.cidr})'
+                },
+                {
+                    'entity': entities.ProvisioningTemplate(),
+                    'entity_type': 'template',
+                },
+                {
+                    'entity': entities.User(),
+                    'value_template': '{entity.login}',
+                    'entity_type': 'user',
+                },
+                {'entity': entities.UserGroup(), 'entity_type': 'usergroup'},
+            ]:
+                created_entity = entity_item['entity'].create()
+                value_template = entity_item.get(
+                    'value_template', '{entity.name}')
+                operation_type = entity_item.get('custom_operation', 'created')
+                entity_value = value_template.format(entity=created_entity)
+                self.audit.filter(
+                    'type={} and action=create'.format(
+                        entity_item['entity_type']
+                    )
+                )
+                result = self.audit.get_last_entry()
```

Comment:
Not sure if that's the best idea since you aren't isolating search results by _specific_ entity. I know tests are marked with `run_in_one_thread`, but it still not 100% guarantee nothing else will be created in the background (like async tasks for failed tests, for example). How about fetching all the entries from the page (instead of the last one) and ensuring our entry is present in the list?

Reply:
not sure what criteria you can use for that isolation. Once more, it is not task log, it is audit log which store only CRUD actions, so no chances to have invalid result for run in one thread action